jmeter/src/main/java/com/excelian/mache/jmeter/mongo/MongoConsoleMultiThreadedTester.java

```diff
@@ -0,0 +1,286 @@
+					mqFactory1, getMQConfiguration(),
+					new MacheFactory<String, MongoTestEntity, Mongo>(),
+					new UUIDUtils());
+			cache1 = cacheFactory1.createCache(new MongoDBCacheLoader<>(
+					MongoTestEntity.class, new CopyOnWriteArrayList<>(Arrays
+							.asList(new ServerAddress(mongoIP, mongoPort))),
+					SchemaOptions.CREATEANDDROPSCHEMA, mongoKeyspace));
+					*/
+		}
+
+		private MQConfiguration getMQConfiguration() {
+			return new MQConfiguration() {
+				@Override
+				public String getTopicName() {
+					return "testTopic";
+				}
+			};
+		}
+		
+
+	    public void teardown() {
+	        //if (cache1 != null) cache1.close();
+	        if (mqFactory1 != null) mqFactory1.close();
+	    }
+	}
+
+	static class WriterCallable extends MongoBaseCallable implements
+			Callable<Statistics> {
+		private static AtomicReference<String> lastWrittenValue = new AtomicReference<String>(
+				null);
+		private final int writerSleepMs;
+		private final String entityKey;
+
+		public WriterCallable(int writerSleepMs, String entityKey,
+				final String mongoIP, final int mongoPort,
+				final String activeMq, final String mongoKeyspace)
+				throws JMSException {
+			super(mongoIP, mongoPort, activeMq, mongoKeyspace);
+			this.writerSleepMs = writerSleepMs;
+			this.entityKey = entityKey;
+
+		}
+
+		public static String getLastWrittenValue() {
+			return lastWrittenValue.get();
+		}
+
+		@Override
+		public Statistics call() throws InterruptedException {
+			final Statistics result = new Statistics();
+			result.milisStarted = System.currentTimeMillis();
+
+			for (int i = 0; i < 20000000; ++i) {
+				if (!Thread.interrupted()) {
+					String nValue = generateValue();
+					MongoTestEntity tE = new MongoTestEntity(this.entityKey,
+							nValue);
+					//cache1.put(this.entityKey, tE);
+					lastWrittenValue.set(nValue);
+					result.runsCount++;
+					result.okCount++;
+					Thread.sleep(writerSleepMs);
+				}
+			}
+
+			result.milisEnded = System.currentTimeMillis();
+			
+			teardown();
+			return result;
+		}
+
+		private String generateValue() {
+			return Generators.timeBasedGenerator().generate().toString();
+		}
+	}
+
+	static class ReaderCallable extends MongoBaseCallable implements Callable<Statistics> {
+		private final int readerSleepMs;
+		private final int readerTimeoutMs;
+		private final String entityKey;
+
+		public ReaderCallable(int readerSleepMs, int readerTimeoutMs,
+				String entityKey, final String mongoIP, final int mongoPort,
+				final String activeMq, final String mongoKeyspace)
+				throws JMSException {
+			super(mongoIP, mongoPort, activeMq, mongoKeyspace);
+			this.readerSleepMs = readerSleepMs;
+			this.readerTimeoutMs = readerTimeoutMs;
+			this.entityKey = entityKey;
+		}
+
+
+		@Override
+		public Statistics call() throws InterruptedException {
+			while(WriterCallable.getLastWrittenValue() == null) {
+				Thread.sleep(0);
```

Comment:
is this effectively to just get this thread to yield? - if so perhaps a call to yield would be better?